tests/test_pedigree.py

```diff
@@ -174,3 +175,153 @@ def test_pedigree_sanity_checks(self):
 
         with pytest.raises(ValueError):
             ped.get_times(individual=ped.individual)
+
+
+def simulate_pedigree(
+    n_founders=8,
+    n_children_prob=None,
+    n_generations=3,
+    random_sex=True,
+) -> tskit.IndividualTable:
+    def choose_sex(random=True, sex=None):
+        # male = 1, female = 2
+        if random:
+            sex = np.random.binomial(n=1, p=0.5) + 1
```

Comment:
By using ``np.random`` methods we're affecting the global state in the numpy module, which we don't want to do for test cases (test cases should always be self contained and deterministic). So, we should do something like 
```
rng = np.random.RandomState(random_seed)
...
sex = rng.binomial(n=1, p=0.5) + 1
```